components/ILIAS/AuthApache/src/AuthFrontendCredentialsApache.php

```diff
@@ -94,46 +103,51 @@ protected function getSettings(): ilSetting
 
     public function initFromRequest(): void
     {
-        $mappingFieldName = $this->getSettings()->get('apache_auth_username_direct_mapping_fieldname', '');
+        $mapping_field_name = $this->getSettings()->get('apache_auth_username_direct_mapping_fieldname', '');
 
-        $this->logger->dump($this->httpRequest->getServerParams(), ilLogLevel::DEBUG);
-        $this->logger->debug($mappingFieldName);
+        $this->logger->dump($this->http_request->getServerParams(), ilLogLevel::DEBUG);
+        $this->logger->debug($mapping_field_name);
 
         switch ($this->getSettings()->get('apache_auth_username_config_type')) {
-            case ilAuthProviderApache::APACHE_AUTH_TYPE_DIRECT_MAPPING:
-                if (isset($this->httpRequest->getServerParams()[$mappingFieldName])) {
-                    $this->setUsername($this->httpRequest->getServerParams()[$mappingFieldName]);
+            case AuthProviderApache::APACHE_AUTH_TYPE_DIRECT_MAPPING:
+                if (isset($this->http_request->getServerParams()[$mapping_field_name])) {
+                    $this->setUsername($this->http_request->getServerParams()[$mapping_field_name]);
                 }
                 break;
 
-            case ilAuthProviderApache::APACHE_AUTH_TYPE_BY_FUNCTION:
-                $this->setUsername(ApacheCustom::getUsername());
+            case AuthProviderApache::APACHE_AUTH_TYPE_BY_FUNCTION:
+                $factory = new UsernameProviderFactory();
+                $resolver = new UsernameResolver($factory->fromClassNames(
+                    require CollectUsernameProvidersObjective::PATH()
```

Comment:
```suggestion
                    require_once CollectUsernameProvidersObjective::PATH()
```

Souldn't we make sure to require the providers file just once?

Reply:
If we assume that `\ILIAS\AuthApache\AuthFrontendCredentialsApache::initFromRequest` is only called once per request, `require_once` should be fine here.

Please note that require an artifact here, which is a PHP file returning an array. We don't require a class here.